src/SensioLabs/Behat/PageObjectExtension/PageObject/Factory/CamelcasedClassNameResolver.php

```diff
@@ -58,15 +58,18 @@ private function resolve($pageObjectName, array $namespaces, $pageObjectType)
         $classNameCandidates = $this->getClassNameCandidates($namespaces, $pageObjectName);
         $className = $this->findExistingClassName($classNameCandidates);
 
-        if (null === $className) {
-            $message = sprintf('Could not find a class for the "%s" %s. ', $pageObjectName, $pageObjectType);
-            $message.= sprintf('None of the configured namespaces worked: "%s"', implode($classNameCandidates, ', '));
+        if (null !== $className) {
+            return $className;
+        }
 
-            throw new \InvalidArgumentException($message);
+        if ($this->isPageObjectNameFQCN($pageObjectName)) {
```

Comment:
Would it be possible to move this condition to the beginning of the method? We wouldn't have to evaluate other parts of this method if the passed argument is already a class name.